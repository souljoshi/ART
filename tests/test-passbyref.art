int set(int &x, int v)
{
    x = v;
}

int xchg(int &x1, int &x2)
{
    int tmp;
    tmp = x1;
    x1 = x2;
    x2 = tmp;
}

int no_change(int x)
{
    x = 322;
}

int main ()
{
    int a, b, c;
    a = 0;
    b = 3;
    c = 4;

    printi(a);
    printi(b);
    printi(c);

    no_change(a); 
    no_change(b);
    no_change(c);

    printc('\n');
    printi(a);
    printi(b);
    printi(c);

    set (a, 7);
    xchg (b, c);

    printc('\n');
    printi(a);
    printi(b);
    printi(c);
}