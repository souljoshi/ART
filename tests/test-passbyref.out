0
3
4

0
3
4

7
4
3
